cluster/mesos/docker/util.sh

```diff
@@ -0,0 +1,317 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Example:
+# export KUBERNETES_PROVIDER=mesos/docker
+# go run hack/e2e.go -v -up -check_cluster_size=false
+# go run hack/e2e.go -v -test -check_version_skew=false
+# go run hack/e2e.go -v -down
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(cd "$(dirname "${BASH_SOURCE}")/../../.." && pwd)
+provider_root="${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}"
+
+source "${provider_root}/${KUBE_CONFIG_FILE-"config-default.sh"}"
+source "${KUBE_ROOT}/cluster/common.sh"
+
+
+# Run kubernetes scripts inside docker.
+# This bypasses the need to set up network routing when running docker in a VM (e.g. boot2docker).
+# Trap signals and kills the docker container for better signal handing
+function cluster::mesos::docker::run_in_docker {
```

Comment:
The ones without the prefix are re-implementation of `cluster/kube-utils.sh` functions.